subprojects/wrapper/src/main/java/org/gradle/wrapper/Download.java

```diff
@@ -55,12 +56,84 @@ public Download(Logger logger, DownloadProgressListener progressListener, String
         configureProxyAuthentication();
     }
 
-    private void configureProxyAuthentication() {
-        if (System.getProperty("http.proxyUser") != null) {
-            Authenticator.setDefault(new ProxyAuthenticator());
+    private void populateProxyPropertiesFromEnvironmentVariable(String protocol) {
+        // Do not fallback to environment variables if a proxy has been set via properties.
+        if (System.getProperty(protocol + ".proxyHost") != null) {
+            return;
+        }
+
+        String variableName = protocol + "_proxy";
+        String proxyString = System.getenv(variableName);
+        if (proxyString == null) {
+            proxyString = System.getenv(variableName.toUpperCase());
+        }
```

Comment:
❌ So the environmental variables we read here are `HTTP_PROXY` and `HTTPS_PROXY`. I think we should not calculate the key, but explicitly define them as a static final variable.

Also, do we actually need this? It seems like that reading the system properties is enough, or do I miss a use-case?

Reply:
The use-case is if people behind a proxy start to use Gradle. Such users usually already have the `http_proxy` / `https_proxy` environment variables defined for other programs, like cURL, to work. Consequently, these users often expected (with reason) that Gradle should also work, but it doesn't without this change.

Reply:
It makes sense. It seems to me though, that we should add documentation explaining the expected behavior.